sdk/keyvault/azure-security-keyvault-keys/test/ut/key_rotation_policy_test_live.cpp

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "gtest/gtest.h"
+
+#include "../../src/private/key_serializers.hpp"
+#include "key_client_base_test.hpp"
+#include <azure/core/base64.hpp>
+#include <azure/core/datetime.hpp>
+#include <azure/keyvault/keyvault_keys.hpp>
+#include <private/key_constants.hpp>
+
+#include <string>
+
+using namespace Azure::Security::KeyVault::Keys::Test;
+using namespace Azure;
+using namespace Azure::Security::KeyVault::Keys;
+using namespace Azure::Security::KeyVault::Keys::_detail;
+
+TEST_F(KeyVaultKeyClient, GetKeyRotationPolicy)
+{
+  auto const keyName = GetTestName();
+  auto const& client = GetClientForTest(keyName);
+
+  auto createKeyResponse = client.CreateEcKey(CreateEcKeyOptions(keyName));
+  CheckValidResponse(createKeyResponse);
+
+  std::string input
+      = "{\"id\":\"https://redacted.vault.azure.net/keys/GetKeyRotationPolicy/"
+        "rotationpolicy\",\"lifetimeActions\":[{\"trigger\":{\"timeAfterCreate\":\"P18M\"},"
+        "\"action\":{\"type\":\"Rotate\"}},{\"trigger\":{\"timeBeforeExpiry\":\"P30D\"},\"action\":"
+        "{\"type\":\"Notify\"}}],\"attributes\":{\"expiryTime\":\"P48M\",\"created\":1649797765,"
+        "\"updated\":1649797765}}";
+
+  auto policy = KeyRotationPolicySerializer::KeyRotationPolicyDeserialize(
+      std::vector<uint8_t>(input.begin(), input.end()));
+
+  auto putPolicy = client.PutKeyRotationPolicy(keyName, policy).Value;
+  auto rotationPolicy = client.GetKeyRotationPolicy(keyName).Value;
+
+  EXPECT_EQ(rotationPolicy.Attributes.ExpiryTime.Value(), policy.Attributes.ExpiryTime.Value());
+  EXPECT_NE(rotationPolicy.Id.size(), size_t(0));
+  EXPECT_EQ(rotationPolicy.LifetimeActions.size(), policy.LifetimeActions.size());
+
+  for (auto result : rotationPolicy.LifetimeActions)
+  {
+    bool found = false;
+
+    for (auto original : policy.LifetimeActions)
+    {
+      if (result.Action == original.Action)
+      {
+        found = true;
+        break;
+      }
+    }
+    EXPECT_TRUE(found);
+  }
+}
+
+TEST_F(KeyVaultKeyClient, DISABLED_GetRandomBytes)
+{ // NEED TO DISABLE TEST FOR THE MOMENT.
+  // DUE TO ISSUE WITH CREATE EC HSM TEST WHICH FAILS WITH ACTUAL HSM BEING SET IN THE ENVIRONMENT
+  // VARIABLE FILED BUG 3563 TO FIX IT
```

Comment:
Create an issue to keep track of re-enabling this test

Reply:
issue 3563